src/NServiceBus.Core/Transports/TransportReceiver.cs

```diff
@@ -61,19 +64,42 @@ class TransportReceiver
 
         async Task InvokePipeline(PushContext pushContext)
         {
+            //note: we should get this value from the transport in the future to make it more accurate
+            var pipelineStartedAt = DateTime.UtcNow;
+
+
             using (var childBuilder = builder.CreateChildBuilder())
             {
                 var rootContext = new RootContext(childBuilder, pipelineCache, eventAggregator);
-                var context = new TransportReceiveContext(new IncomingMessage(pushContext.MessageId, pushContext.Headers, pushContext.BodyStream), pushContext.TransportTransaction, pushContext.ReceiveCancellationTokenSource, rootContext);
+
+                //notifications are only relevant for the main pipeline since satellites pipelines are not exposed to user in any way
+                if (isMainReceiver)
+                {
+                    var sw = new Stopwatch();
+
+                    sw.Start();
+
+                    // float the sw via the context to avoid threading issues should the continuation where we access 
+                    // it is executed on a different thread
+                    rootContext.Set(sw);
```

Comment:
I've read the article about it in here: https://msdn.microsoft.com/en-us/library/dn553408%28v=vs.85%29.aspx
The most meaningful from our perspective is the following, which according to my understanding can be summarized as 'we should not care about this limited skew'.
>In general, the performance counter results are consistent across all processors in multi-core and multi-processor systems, even when measured on different threads or processes. Here are some exceptions to this rule:
- Pre-Windows Vista operating systems that run on certain processors might violate this consistency because of one of these reasons:
 -   The hardware processors have a non-invariant TSC and the BIOS doesn't indicate this condition correctly.
 -        The TSC synchronization algorithm that was used wasn't suitable for systems with large numbers of processors.
-    When you compare performance counter results that are acquired from different threads, consider values that differ by ± 1 tick to have an ambiguous ordering. If the time stamps are taken from the same thread, this ± 1 tick uncertainty doesn't apply. In this context, the term tick refers to a period of time equal to 1 ÷ (the frequency of the performance counter obtained from QueryPerformanceFrequency).